client/state/draft-feedback/reducer.js

```diff
@@ -0,0 +1,71 @@
+/**
+ * Internal dependencies
+ */
+import {
+	EDITOR_START,
+	EDITOR_STOP,
+	DRAFT_FEEDBACK_SHARE_ADD,
+	DRAFT_FEEDBACK_SHARE_REMOVE,
+	DRAFT_FEEDBACK_SHARE_REVOKE,
+	DRAFT_FEEDBACK_SHARE_RESTORE,
+	DRAFT_FEEDBACK_COMMENT_ADD,
+} from 'state/action-types';
+import { keyedReducer, withoutPersistence } from 'state/utils';
+
+const initialState = {};
+const initialDraftShareState = {
+	enabled: true,
+	comments: [],
+};
+
+const draftShareReducer = keyedReducer(
+	'emailAddress',
+	( state = initialDraftShareState, action ) => {
+		switch ( action.type ) {
+			case DRAFT_FEEDBACK_SHARE_ADD:
+				return initialDraftShareState;
+
+			case DRAFT_FEEDBACK_SHARE_REMOVE:
+				return undefined;
+
+			case DRAFT_FEEDBACK_SHARE_REVOKE:
+				return {
+					...state,
+					enabled: false,
+				};
+
+			case DRAFT_FEEDBACK_SHARE_RESTORE:
+				return {
+					...state,
+					enabled: true,
+				};
+
+			case DRAFT_FEEDBACK_COMMENT_ADD:
+				return {
+					...state,
+					comments: state.comments.concat( action.comment ),
+				};
+
+			default:
+				return state;
+		}
+	},
+);
+
+export default withoutPersistence(
+	keyedReducer(
+		'siteId',
+		keyedReducer( 'postId', ( state, action ) => {
+			switch ( action.type ) {
+				case EDITOR_START:
+					return {};
+
+				case EDITOR_STOP:
+					return undefined;
+
+				default:
+					return draftShareReducer( state, action );
+			}
+		} ),
+	),
+);
```

Comment:
I see what you are doing here, but I wonder if we might be better off with some wrapper around the keyed reducers instead of inside of them. we have had discussions about gating the keyed reducers and acting on more broad action types (for example, one which should affect every item in the collection) and our conclusion at the time was that it makes most sense to do this from a composed wrapper instead of embedding the logic.

what would that look like here?

well I think that this is confusing because in some cases it looks like we are using a reducer and in others not. what if instead we passed the draft share reducer into the other…

```js
export const withEditor = reducer => ( state, action ) => {
	switch ( action.type ) {
		case EDITOR_START:
			return {};

		case EDITOR_STOP:
			return undefined;

		default:
			reducer( state, action );
	}
}

export const draftShares = keyedReducer( 'siteId', keyedReducer( 'postId', withEditor( draftShare ) ) );
```

I don't know how much better this is than yours and I still wonder if it should work at a higher level, but it's an idea. You can think about it and make your own decisions.